### PR TITLE
[pull-request:create] Allow to define choices in PR table

### PR DESCRIPTION
**Pull request by phansys**
|Q            |A     |
|---          |---   |
|Branch       |master|
|Bug fix?     |no    |
|New feature? |yes   |
|BC breaks?   |no    |
|Deprecations?|no    |
|Tests pass?  |yes   |
|Fixed tickets|      |
|License      |MIT   |
|Doc PR       |      |

**TODO:**
~~- [ ] Improve tests, mainly the logic related to validator.~~

### CODE REVIEW

**Comment by sstok** [comment]
I think it's easier to mark a question as a choice/confirm, and also making it possible to configure this on the pr-template configuration.

```
table-pr:
    branch: ['Branch', master]
    bug_fix: ['Bug fix?', no, confirm]
    new_feature: ['New feature?', no, confirm]
    bc_breaks: ['BC breaks?', no, confirm]
    deprecations: ['Deprecations?', no, confirm]
    tests_pass: ['Tests pass?', no, confirm]
    fixed_tickets: ['Fixed tickets', '']
    license: ['License', MIT, ['MIT', 'BSD-3']]
```

File: src/Helper/TemplateHelper.php [hunk 1]
```diff
@@ -150,6 +151,20 @@ public function getTemplate($domain, $name)
     public function parameterize(TemplateInterface $template)
     {
         $params = [];
+        $validYesNoResponses = ['yes', 'y', 'no', 'n'];
```
Comment (cordoval) [inline src/Helper/TemplateHelper.php]: move this if at all to a Validator folder, also i rather have this to work by default if any character is pressed it should be yes, if nothing then no. Or similar.

File: tests/Template/PullRequest/Create/SymfonyTemplateTest.php [hunk 2]
```diff
@@ -85,13 +115,27 @@ public function provideTemplate()
      */
     public function runs_template_command_with_symfony_template($params, $expected)
     {
+        $validYesNoResponses = ['yes', 'y', 'no', 'n'];
```
Comment (cordoval) [inline tests/Template/PullRequest/Create/SymfonyTemplateTest.php]: refactor as per above note

**Comment by phansys** [comment]
I prefer to left the validation stuff to another PR. Please share any thoughts about the proposed choice syntax. Any comment is welcome.

**Comment by cordoval** [comment]
@phansys just a note, glad that you did not change the behavior because i actually like that you can just type 'n' and not the full no word.

**Pull request merged**